catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/cache/solr/impl/CachingFederationStrategyTest.java

```diff
@@ -0,0 +1,328 @@
+        killAndWaitForExecutor();
+    }
+
+    @Test
+    public void testFederateQueryCache() throws Exception {
+
+        HashMap<String, Serializable> properties = new HashMap<>();
+        properties.put(QUERY_MODE, CACHE_QUERY_MODE);
+
+        QueryRequest fedQueryRequest = new QueryRequestImpl(mockQuery, properties);
+
+        MockSource mockSource = getMockSource(mockResponse);
+
+        List<Source> sources = Arrays.asList(mockSource);
+
+        QueryResponse response = strategy.federate(sources, fedQueryRequest);
+
+        WaitCondition.expect("Wait until the result appears in the response")
+                .until(() -> response.getHits(), is(HITS_RESULT))
+                .checkEvery(POLLING_INTERVAL, TimeUnit.MILLISECONDS)
+                .within(WITHIN_TIMEOUT, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void testFedereateQueryCacheNoUpdate() throws Exception {
+        HashMap<String, Serializable> properties = new HashMap<>();
+        properties.put(QUERY_MODE, NATIVE_QUERY_MODE);
+
+        QueryRequest fedQueryRequest = new QueryRequestImpl(mockQuery, properties);
+
+        MockSource mockSource = getMockSource(mockResponse);
+
+        List<Source> sources = Arrays.asList(mockSource);
+
+        QueryResponse response = strategy.federate(sources, fedQueryRequest);
+
+        WaitCondition.expect("Wait until the result appears in the response")
+                .until(() -> response.getHits(), is(HITS_RESULT))
+                .checkEvery(POLLING_INTERVAL, TimeUnit.MILLISECONDS)
+                .within(WITHIN_TIMEOUT, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void testFederateQueryAndUpdateCacheBlocking() throws Exception {
+        HashMap<String, Serializable> properties = new HashMap<>();
+        properties.put(QUERY_MODE, INDEX_QUERY_MODE);
+
+        QueryRequest fedQueryRequest = new QueryRequestImpl(mockQuery, properties);
+
+        MockSource mockSource = getMockSource(mockResponse);
+
+        List<Source> sources = Arrays.asList(mockSource);
+
+        QueryResponse response = strategy.federate(sources, fedQueryRequest);
+
+        WaitCondition.expect("Wait until the result appears in the response")
+                .until(() -> response.getHits(), is(HITS_RESULT))
+                .checkEvery(POLLING_INTERVAL, TimeUnit.MILLISECONDS)
+                .within(WITHIN_TIMEOUT, TimeUnit.SECONDS);
+    }
+
+    @Test
+    public void testFederateQueryAndUpdateCacheNotBlocking() throws Exception {
+        HashMap<String, Serializable> properties = new HashMap<>();
+        properties.put(QUERY_MODE, UPDATE_QUERY_MODE);
+
+        QueryRequest fedQueryRequest = new QueryRequestImpl(mockQuery, properties);
+
+        MockSource mockSource = getMockSource(mockResponse);
+
+        List<Source> sources = Arrays.asList(mockSource);
+
+        QueryResponse response = strategy.federate(sources, fedQueryRequest);
+
+        WaitCondition.expect("Wait until the result appears in the response")
```

Comment:
Same as above, should we check that the result came from a different source and that the cache was updated?